pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -87,5 +142,51 @@ var _ = Describe("Qemu agent poller", func() {
 				Id:            "testguestos"}
 			Expect(guestOSInfoStatus).To(Equal(expectedGuestOSInfo))
 		})
+
+		Context("with AgentStore", func() {
+
+			It("should fire an event for new sysinfo data", func() {
+				var agentStore = NewAgentAstore()
+				fakeInfo := api.GuestOSInfo{
+					Name:          "TestGuestOSName",
+					KernelRelease: "1.1.0-Generic",
+					Version:       "1.0.0",
+					PrettyName:    "TestGuestOSName 1.0.0",
+					VersionId:     "1.0.0",
+					KernelVersion: "1.1.0",
+					Machine:       "x86_64",
+					Id:            "testguestos",
+				}
+				agentStore.Store(GET_OSINFO, fakeInfo)
+
+				Expect(agentStore.AgentUpdated).To(Receive(Equal(AgentUpdatedEvent{
+					Type:       GET_OSINFO,
+					DomainInfo: api.DomainGuestInfo{OSInfo: &fakeInfo},
+				})))
+			})
+
+			It("should not fire an event for the same sysinfo data", func() {
+				var agentStore = NewAgentAstore()
+				fakeInfo := api.GuestOSInfo{
+					Name:          "TestGuestOSName",
+					KernelRelease: "1.1.0-Generic",
+					Version:       "1.0.0",
+					PrettyName:    "TestGuestOSName 1.0.0",
+					VersionId:     "1.0.0",
+					KernelVersion: "1.1.0",
+					Machine:       "x86_64",
+					Id:            "testguestos",
+				}
+
+				agentStore.Store(GET_OSINFO, fakeInfo)
+				Expect(agentStore.AgentUpdated).To(Receive(Equal(AgentUpdatedEvent{
```

Comment:
This check is redundant (see above). Do you do this because the channel has to be polled? Or can we just skip the event somehow?

Reply:
This check have to be present. The second poll is required to see whether the event has not been fired, and this is done via additional poll.

Reply:
I understand that the polling is required. I'm just complaining about the duplicate check. Well, that's just nitpicking so nevermind.